src/rgw/rgw_auth_s3.cc

```diff
@@ -919,25 +908,13 @@ AWSv4ComplMulti::ChunkMeta::create_next(CephContext* const cct,
 std::string
 AWSv4ComplMulti::calc_chunk_signature(const std::string& payload_hash) const
 {
-  constexpr size_t algorithm_len = std::strlen(AWS4_HMAC_SHA256_STR);
-  constexpr size_t empty_hash_len = std::strlen(AWS4_EMPTY_PAYLOAD_HASH);
-
-  /* We want to avoid reallocations when concatenating the string_to_sign. */
-  const size_t total_len = algorithm_len + date.length() + \
-    credential_scope.length() + prev_chunk_signature.length() + \
-    empty_hash_len + payload_hash.length() + std::strlen("\n") * 5;
-  const auto string_to_sign = create_n_reserve<std::string>(total_len)
-    .append(AWS4_HMAC_SHA256_STR, algorithm_len)
-    .append("\n", std::strlen("\n"))
-    .append(date.data(), date.length())
-    .append("\n", std::strlen("\n"))
-    .append(credential_scope.data(), credential_scope.length())
-    .append("\n", std::strlen("\n"))
-    .append(prev_chunk_signature)
-    .append("\n", std::strlen("\n"))
-    .append(AWS4_EMPTY_PAYLOAD_HASH, empty_hash_len)
-    .append("\n", std::strlen("\n"))
-    .append(payload_hash);
+  const auto string_to_sign = string_join_reserve("\n",
```

Comment:
i'm not sure it's actually faster, as it was already calculating the total size for `string::reserve()`. i just think the variadic version is less error-prone by doing the size calculation for you